src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt

```diff
@@ -27,56 +23,15 @@ class CellUpdater: Updater {
-                    cell.speed += oppositeForce * hardnessCoefficient * delta
-                }
+                clashCellWithBorders(oldCell, border, cell, delta)
             }
 
             for (other in oldArea.cells.values) {
-                if (other.id == cell.id) continue
-                if (cell.center.distance(other.center) <= cell.radius + other.radius) {
-                    val intersections = testCirclesIntersection(
-                        oldCell.center, oldCell.radius,
-                        other.center, other.radius
-                    )
-
-                    val pivot =
-                        if (intersections.isNotEmpty())
-                            (intersections.first() + intersections.last()) / 2
-                        else
-                            (oldCell.center + other.center) / 2
-
-                    val depth = (1 - oldCell.center.distance(pivot)/oldCell.radius).coerceIn(0.0, 1.0)
-
-                    val massSum = oldCell.mass + other.mass
-                    val thisMassCoefficient = oldCell.mass / massSum
-                    val oppositeForce = (pivot to oldCell.center).unitSafe() * (oldCell.radius - pivot.distance(oldCell.center))
-                    val hardnessCoefficient = (oldCell.genome.hardness*depth + 1).pow(oldCell.genome.hardness + 1)
-                    cell.speed += (oppositeForce * hardnessCoefficient + (other.speed * thisMassCoefficient - oldCell.speed * (1 - thisMassCoefficient)) * oldCell.genome.hardness) * delta
-                }
+                clashCells(other, cell, oldCell, delta)
             }
 
             for (connection in oldCell.connections.values) {
-                val partner = oldArea.cells[connection.partnerId]
-                if (partner != null) {
-                    val partnerConnection = partner.connections[cell.id]
-                    if (partnerConnection != null) {
-                        repeat(4) { stringId ->
-                            val angleOffset = stringId* PI /24 - PI /12
-                            val effectiveConnectionAngle = oldCell.angle + connection.angle + angleOffset
-                            val effectiveConnectionForceOrigin = oldCell.center + Vector2.unit(effectiveConnectionAngle) * oldCell.radius - Vector2.unit(effectiveConnectionAngle) * CELL_STICKINESS_DEPTH
-                            val effectiveConnectionForceDestination = partner.center + Vector2.unit(partner.angle + partnerConnection.angle - angleOffset) * partner.radius - Vector2.unit(partner.angle + partnerConnection.angle - angleOffset) * CELL_STICKINESS_DEPTH
-                            val connectionForceDirection = (effectiveConnectionForceOrigin to effectiveConnectionForceDestination) / 4 * delta
-                            applyImpulse(oldCell, cell, effectiveConnectionForceOrigin, connectionForceDirection)
-                        }
-                    }
-                }
+                moveCell(oldArea, connection, cell, oldCell, delta)
```

Comment:
I think the name "moveCell" could be unclear as the function is doing only cells connection-related physics

Reply:
fixed